cpp/src/cylon/column.hpp

```diff
@@ -19,42 +19,95 @@
 #include <utility>
 #include <memory>
 #include "data_types.hpp"
+#include "arrow/arrow_types.hpp"
 
 #include <arrow/api.h>
 #include <arrow/table.h>
 
+#include <iostream>
+
 namespace cylon {
 
 class Column {
  public:
-  Column(std::string id, std::shared_ptr<DataType> type) : id(std::move(id)), type(std::move(type)) {
+  Column(const std::string &id, const std::shared_ptr<DataType> &type,
+         const std::shared_ptr<arrow::ChunkedArray> &data_)
+      : id(id), type(type), data_(data_) {
+  }
+
+  Column(const std::string &id, const std::shared_ptr<DataType> &type,
+         const std::shared_ptr<arrow::Array> &data_)
+      : id(id), type(type),
+        data_(std::make_shared<arrow::ChunkedArray>(data_)) {
   }
 
   /**
-   * From the arrow array
-   * @param array the array to use
-   * @return the column
+   * Return the data wrapped by column 
+   * @return arrow chunked array
    */
-  std::shared_ptr<Column> FromArrow(std::shared_ptr<arrow::Array> array);
+  std::shared_ptr<arrow::ChunkedArray> GetColumnData() const;
 
   /**
    * Return the unique id of the array
    * @return
    */
-  std::string get_id() {
-	return this->id;
-  }
+  std::string GetID() const;
 
- private:
   /**
-   * The id of the column
+   * Return the data type of the column
+   * @return
    */
-  std::string id;
+  std::shared_ptr<DataType> GetDataType() const;
 
-  /**
-   * The datatype of the column
-   */
-  std::shared_ptr<DataType> type;
+  static std::shared_ptr<Column> Make(const std::string &id, const std::shared_ptr<DataType> &type,
+                                      const std::shared_ptr<arrow::ChunkedArray> &data_);
+
+  static std::shared_ptr<Column> Make(const std::string &id, const std::shared_ptr<DataType> &type,
+                                      const std::shared_ptr<arrow::Array> &data_);
+
+ private:
+  std::string id; // The id of the column
+  std::shared_ptr<DataType> type; // The datatype of the column
+
+ protected:
+  Column(std::string id, std::shared_ptr<DataType> type)
+      : id(std::move(id)), type(std::move(type)) {
+  }
+
+  std::shared_ptr<arrow::ChunkedArray> data_;   // pointer to the data array
+};
+
+template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
+class VectorColumn : public Column {
+ public:
+  VectorColumn(const std::string &id,
+               const std::shared_ptr<DataType> &type,
+               const std::shared_ptr<std::vector<T>> &data_vector) :
```

Comment:
Who owns this vector? Wha will happen if this vector goes out of scope and its memory is freed?